bcc/module.go

```diff
@@ -274,6 +285,37 @@ func (bpf *Module) AttachTracepoint(name string, fd int) error {
 	return nil
 }
 
+func (bpf *Module) AttachPerfEvent(evType, evConfig int, samplePeriod int, sampleFreq int, pid, cpu, groupFd, fd int) error {
+	key := fmt.Sprintf("%d:%d", evType, evConfig)
+	if _, ok := bpf.perfEvents[key]; ok {
+		return nil
+	}
+
+	res := make([]int, runtime.NumCPU())
+
+	if cpu > 0 {
+		r, err := C.bpf_attach_perf_event(C.int(fd), C.uint(evType), C.uint(evConfig), C.ulong(samplePeriod), C.ulong(sampleFreq), C.int(pid), C.int(cpu), C.int(groupFd))
+		if r < 0 {
+			return fmt.Errorf("failed to attach BPF perf event: %v", err)
+		}
+
+		res[cpu] = int(r)
+	} else {
+		for i := 0; i < len(res); i++ {
```

Comment:
This doesn't work when not all CPUs are online. For example, after disabling one CPU out of four:

```
# echo 0 > /sys/devices/system/cpu/cpu1/online
# cat /sys/devices/system/cpu/online 
0,2-3
```

```
perf_event_open failed: No such device
Failed to attach perf event: failed to attach BPF perf event: no such device
```

You can use the [cpuonline](https://github.com/iovisor/gobpf/blob/master/pkg/cpuonline/cpu_range.go) package instead of `runtime.NumCPU()`.